### ISSUE TITLE
New Rule: Vercel API Token (@secretlint/secretlint-rule-vercel)

### ISSUE THREAD

**Issue by azu** [secretlint/secretlint#1450]
## Description

Add a new rule to detect Vercel API tokens.

Vercel introduced new prefixed token formats with 5 distinct types, each requiring a separate message:

- `vcp_` — Personal access token
- `vci_` — Integration token
- `vca_` — App access token (OAuth)
- `vcr_` — App refresh token (OAuth)
- `vck_` — API key (AI Gateway)

## Source

- [Vercel New Token Formats announcement](https://vercel.com/changelog/new-token-formats-and-secret-scanning)
- [Vercel OAuth Tokens documentation](https://vercel.com/docs/sign-in-with-vercel/tokens)

Note: Vercel is not yet listed in GitHub Secret Scanning Push Protection, but the token format is well-defined and detectable with low false positive rate.

### VERCEL_PERSONAL_ACCESS_TOKEN
Prefix `vcp_` followed by 20–60 alphanumeric characters.

### VERCEL_INTEGRATION_TOKEN
Prefix `vci_` followed by 20–60 alphanumeric characters.

### VERCEL_APP_ACCESS_TOKEN
Prefix `vca_` followed by 20–60 alphanumeric characters.

### VERCEL_APP_REFRESH_TOKEN
Prefix `vcr_` followed by 20–60 alphanumeric characters.

### VERCEL_AI_GATEWAY_API_KEY
Prefix `vck_` followed by 20–60 alphanumeric characters.

Regex (with Unicode boundary, ReDoS-safe):
```
/(?<!\p{L})(vcp|vci|vca|vcr|vck)_[A-Za-z0-9]{20,60}/gu
```

## OK (should NOT be detected)

```
# placeholder / too short
vcp_abc
```

## NG (should be detected)

```
# VERCEL_PERSONAL_ACCESS_TOKEN
VERCEL_TOKEN=vcp_aBcDeFgHiJkLmNoPqRsT1234567890abcdefghij

# VERCEL_APP_ACCESS_TOKEN
access_token: "vca_BQuu9ChDu3n6Pfh6YQnCshpoYkWDSFKogLqmBtQ0tC8NAA5rXt340sjz"
```

**Comment by azu** [comment]
https://github.com/secretlint/secretlint/pull/1451
